src/db-service.ts

```diff
@@ -0,0 +1,164 @@
+// global
+import { Item } from 'graasp';
+import { sql, DatabaseTransactionConnectionType as TrxHandler } from 'slonik';
+// local
+
+/**
+ * Database's first layer of abstraction for advanced search features
+ */
+export class SearchService{
+  publishedTagId: string;
+
+  constructor(publishedTagId: string){
+    this.publishedTagId = publishedTagId;
+  }
+
+  private static allColumns = sql.join(
+    [
+      'id',
+      'name',
+      'description',
+      'type',
+      'path',
+      'extra',
+      'settings',
+      'creator',
+      ['created_at', 'createdAt'],
+      ['updated_at', 'updatedAt'],
+    ].map((c) =>
+      !Array.isArray(c)
+        ? sql.identifier([c])
+        : sql.join(
+            c.map((cwa) => sql.identifier([cwa])),
+            sql` AS `,
+          ),
+    ),
+    sql`, `,
+  );
+
+  private static allColumnsForJoinedTable = sql.join(
+    [
+      ['item.id', 'id'],
+      'item.name',
+      'description',
+      'item.type',
+      'path',
+      'item.extra',
+      'settings',
+      'creator',
+      ['item.created_at', 'createdAt'],
+      ['item.updated_at', 'updatedAt'],
+    ].map((c) =>
+      !Array.isArray(c)
+        ? sql.identifier([c])
+        : sql.join(
+            c.map((cwa) => sql.identifier([cwa])),
+            sql` AS `,
+          ),
+    ),
+    sql`, `,
+  );
+
+  /**
+  * <keywordSequence>: string, containing formatted keyword. 
+  * @return items containing keywords in name
+  * Note: When searching with Name, the input will be treated as a single keyword
+  */
+  async getItemsMatchName(keywordSequence: string, transactionHandler: TrxHandler): Promise<Item[]> {
+    return (
+      transactionHandler
+        .query<Item>(
+          sql`
+          WITH published_item_paths AS (
+            SELECT item_path FROM item_tag
+            WHERE tag_id = ${this.publishedTagId}
+          )
+          SELECT ${SearchService.allColumns}
+          FROM item
+          WHERE name ILIKE ${keywordSequence}
+            AND path in (SELECT item_path FROM published_item_paths)
+        `,
+        )
+        .then(({ rows }) => rows.slice(0))
+    );
+  }
+
+  /**
+  * <keywordSequence>: string, containing formatted keywords
```

Comment:
I think jsdoc syntax is `@param {string} keywordSequence string containing formatted keywords `. To check.